### PR TITLE
Python exports

### LINKED ISSUE #253: Python exports for new Dataset
The Python bindings are not up-to-date with the major changes to the `Dataset` API. Tags need to be removed. Thanks to that, some things will likely become easier, but nevertheless major parts will need to be rewritten/refactored.

### LINKED ISSUE #266: Remove zip and zipMD and event-list proxies?
It is currently unclear how useful this is, and it would require refactoring to work again with the new `Dataset`.

Should we remove it for now? It adds noise to the code tree and could simply be added back in later.

Keep them only if we intend to re-enable it as part of this release?

### PR DESCRIPTION
**Pull request by SimonHeybrock**
Complete refactor and overhaul of Python exports after the Dataset rewrite and other internal changes on the C++ side.

- Now using `pytest` instead of `unittest`.
- Travis is building `scippy` again and runs the test.
- Some tests have not been refactored yet and are thus disabled.
- The `all` and `install` targets do not build benchmarks any more.
- Split bindings into several modules.
- Using templated binding code for commonalities between various objects and their proxies, greatly reducing the code duplication.
- Finally moved `scipp-core` headers into proper path, avoiding ambiguities when including them in other modules.
- Also removed some legacy files, in particular those related to `zip_md` which is not needed currently and in any case beyond repair due to many of the recent changes, i.e., would need to be rewritten from scratch, if required.

Fixes #266.
Fixes #253.

Not done yet. In particular many of the setters for `Dataset` are missing, as well as operators and free functions (which also do not exist on the C++ side yet). I have opened #283 as a follow-up.

### CODE REVIEW

File: scippy/tests/test_dimensions.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+
+def test_comparison():
+    a = sp.Dimensions(labels=[Dim.X, Dim.Y], shape=[2, 3])
+    a2 = sp.Dimensions(labels=[Dim.X, Dim.Y], shape=[2, 3])
+    b = sp.Dimensions(labels=[Dim.X, Dim.Y], shape=[2, 4])
+    assert a == a
+    assert a == a2
+    assert a != b
+
```
Comment (nvaytet) [inline scippy/tests/test_dimensions.py]: maybe add something like
```
c = sp.Dimensions(labels=[Dim.Y, Dim.X], shape=[3, 2])
assert a != c
```

File: scippy/tests/test_units_neutron.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
+# SPDX-License-Identifier: GPL-3.0-or-later
+# Copyright (c) 2019 Scipp contributors (https://github.com/scipp)
+# @file
+# @author Simon Heybrock
+import scippy as sp
+from scippy import Dim
+
+
+def test_dims():
```
Comment (nvaytet) [inline scippy/tests/test_units_neutron.py]: not sure this should be part of the tests on `units`? Move to `test_dimensions.py`?

File: scippy/tests/test_variable.py [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+def test_create_with_numpy_dtype():
+    var = sp.Variable(labels=[Dim.X], shape=[2], dtype=np.dtype(np.float32))
+    assert var.dtype == sp.dtype.float
+
+
+def test_create_with_variances():
+    assert not sp.Variable(labels=[Dim.X], shape=[2]).has_variances
+    assert not sp.Variable(labels=[Dim.X], shape=[
+                           2], variances=False).has_variances
+    assert sp.Variable(labels=[Dim.X], shape=[2], variances=True).has_variances
```
Comment (nvaytet) [inline scippy/tests/test_variable.py]: Not that it makes much sense, but
what happens to the `variances=True` syntax if you are creating a `Variable` of type bool with a single value and you add a variance that is either `True` or `False`?

File: scippy/bind_data_access.h [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+  }
+};
+
+template <class... Ts> struct as_VariableViewImpl {
+  template <class Getter, class Var>
+  static std::variant<std::conditional_t<
+      std::is_same_v<Var, Variable>, scipp::span<underlying_type_t<Ts>>,
+      VariableView<underlying_type_t<Ts>>>...>
+  get(Var &view) {
+    switch (view.data().dtype()) {
```
[the remark this answers is not in the document]
Comment (SimonHeybrock) [reply scippy/bind_data_access.h]: Indeed. We can probably do better than now, but I don't think we can completely eliminate intervention.

File: scippy/numpy.h [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+  case 0:
+    return copy_flattened_0d(data, proxy);
+  case 1:
+    return copy_flattened_1d(data, proxy);
+  case 2:
+    return copy_flattened_2d(data, proxy);
+  case 3:
+    return copy_flattened_3d(data, proxy);
+  default:
+    throw std::runtime_error("Numpy array has more dimensions than supported "
```
[the remark this answers is not in the document]
Comment (SimonHeybrock) [reply scippy/numpy.h]: There is, using some recursive calls with variadic templates, but unless we need much more than 4 dimensions this is shorted and easier to understand, I think.

File: scippy/tests/test_units_neutron.py (hunk 2 above)
Comment (SimonHeybrock) [reply scippy/tests/test_units_neutron.py]: `test_dimensions.py` tests `class Dimensions`, this is the labels, which are part of `scipp::units`.

File: scippy/tests/test_variable.py (hunk 3 above)
Comment (SimonHeybrock) [reply scippy/tests/test_variable.py]: Unless you specify the `values` argument you should always match the other overload, so this cannot happen, I think?

**Pull request merged**